blocks/blog-posts/blog-posts.js

```diff
@@ -0,0 +1,120 @@
+import {
+  createOptimizedPicture, loadCSS,
+  readBlockConfig,
+  toClassName,
+} from '../../scripts/lib-franklin.js';
+
+const defaultAuthorName = 'Cognizant Netcentric';
+const defaultAuthorTitle = '';
+const defaultAuthorImage = '/icons/nc.svg';
+
+export function buildCard(card) {
+  const {
+    path, title, image, tags, profiles: authorProfile,
+  } = card;
+
+  const cardElement = document.createElement('article');
+  cardElement.classList.add('teaser');
+
+  if (authorProfile.Image === '') authorProfile.Image = defaultAuthorImage;
+
+  cardElement.innerHTML = `
+    <p class="tags">${JSON.parse(tags).join(', ')}</p>
+    <a href="${path}" target="_self" class="teaser-link">
+      <h2 class="teaser-description">
+        ${title}
+      </h2>
+    </a>
+    <div class="authorprofile-container">
+      <div class="authorprofile-image">
+        <div class="nc-image-base">
+            <div class="nc-image-container " itemscope="" itemtype="http://schema.org/ImageObject">
+                <img class="nc-image" src="${authorProfile.Image ?? defaultAuthorImage}" itemprop="contentUrl" alt="" sizes="10vw" />
+            </div>
+        </div>
+      </div>
+      <div class="authorprofile-info">
+          <div class="authorprofile-name">${authorProfile.Name ?? defaultAuthorName}</div>
+          <div class="authorprofile-position">${authorProfile.Title ?? defaultAuthorTitle}</div>
+      </div>
+    </div>`;
+
+  // Width based on max-width set in css
+  const pictureElement = createOptimizedPicture(image, `Image symbolising ${title}`, false, [{ width: '450' }]);
+  if (image && pictureElement) {
+    cardElement.prepend(pictureElement);
+  }
+
+  return cardElement;
+}
+
+export function createCardsList(parent, cards) {
+  const blogList = document.createElement('ul');
+  blogList.classList.add('related-list', 'blog-cards-container');
+
+  cards.forEach((card) => {
+    const blogListItem = document.createElement('li');
+    blogListItem.classList.add('related-list-item');
+
+    blogListItem.append(buildCard(card));
+    blogList.appendChild(blogListItem);
+  });
+
+  parent.appendChild(blogList);
+}
+
+export async function joinArticlesWithProfiles(articles) {
+  const response = await fetch('/profile-blog.json');
+  const json = await response.json();
+
+  Object.values(articles).forEach((value) => {
+    value.profiles = json.data.find((profile) => profile.Name === value.authors) ?? {};
+  });
+
+  return articles;
+}
+
+export async function getArticles(filter = () => true, maxItems = 7, offset = 0) {
+  const response = await fetch('/insights/query-index.json');
+  const json = await response.json();
```

Comment:
should we store the response in a variable to reuse between several `getArticles` calls or can we rely on browser caching here? Probably at least the json parsing could be skipped with a local caching variable

Reply:
I can store the result in a variable, sure. I wonder if anyone has real world experience in the perf difference vs just relying on built-in browser caching - I will check with the team.

Reply:
@nc-andreashaller I suggest we keep that for performance optimization if really needed. I generally agree with you especially if you navigate from blog list to a blog post. On the post we have the related posts again. Can you create a follow up issue?

Reply:
fully agree that we should not optimise before knowing what the impact is, create https://github.com/hlxsites/netcentric/issues/108 to follow up